diff --git a/theano/gpuarray/opt.py b/theano/gpuarray/opt.py
index ed0f877..ab17124 100644
--- a/theano/gpuarray/opt.py
+++ b/theano/gpuarray/opt.py
@@ -1041,14 +1041,14 @@ def local_gpua_advanced_incsubtensor(op, context_name, inputs, outputs):
     if compute_capability >= 2 and x.ndim == 1 and y.ndim == 0:
         x = x.dimshuffle(0, 'x')
         y = y.dimshuffle('x', 'x')
-        ret = GpuAdvancedIncSubtensor1_dev20(
+        ret = GpuAdvancedIncSubtensor1(
             set_instead_of_inc=set_instead_of_inc)(x, y, ilist).dimshuffle(0)
         return ret
     elif compute_capability < 2 or x.ndim != 2 or y.ndim != 2:
         return GpuAdvancedIncSubtensor1(
             set_instead_of_inc=set_instead_of_inc)
     else:
-        return GpuAdvancedIncSubtensor1_dev20(
+        return GpuAdvancedIncSubtensor1(
             set_instead_of_inc=set_instead_of_inc)
 
 
diff --git a/theano/sandbox/cuda/opt.py b/theano/sandbox/cuda/opt.py
index e32370e..05fe8b7 100644
--- a/theano/sandbox/cuda/opt.py
+++ b/theano/sandbox/cuda/opt.py
@@ -1110,7 +1110,7 @@ def local_gpu_advanced_incsubtensor1(node):
 
                 gpu_op = GpuAdvancedIncSubtensor1(**props_dict)
             else:
-                gpu_op = GpuAdvancedIncSubtensor1_dev20(**props_dict)
+                gpu_op = GpuAdvancedIncSubtensor1(**props_dict)
             return [gpu_op(as_cuda_ndarray_variable(x),
                            as_cuda_ndarray_variable(y), *coords)]
 
@@ -1148,7 +1148,7 @@ def local_gpu_advanced_incsubtensor1(node):
             if (compute_capability < 2 or y.ndim != 2 or x.ndim != 2):
                 gpu_op = GpuAdvancedIncSubtensor1(**node.op._props_dict())
             else:
-                gpu_op = GpuAdvancedIncSubtensor1_dev20(**node.op._props_dict())
+                gpu_op = GpuAdvancedIncSubtensor1(**node.op._props_dict())
             return [host_from_gpu(gpu_op(gpu_x, gpu_y, *coords))]
     return False
 
