diff --git a/theano/sandbox/cuda/opt.py b/theano/sandbox/cuda/opt.py
index 158a5f1..c9cf412 100644
--- a/theano/sandbox/cuda/opt.py
+++ b/theano/sandbox/cuda/opt.py
@@ -38,7 +38,7 @@ from theano.sandbox.cuda.basic_ops import (
     GpuElemwise, GpuDimShuffle, GpuReshape, GpuCAReduce,
     gpu_flatten,
     GpuSubtensor, GpuAdvancedSubtensor1,
-    GpuAdvancedIncSubtensor1, GpuAdvancedIncSubtensor1_dev20,
+    GpuAdvancedIncSubtensor1,
     GpuIncSubtensor, gpu_alloc, GpuAlloc, gpu_shape, GpuSplit, GpuAllocEmpty)
 
 from theano.sandbox.cuda.type import CudaNdarrayType
@@ -1134,7 +1134,8 @@ def local_gpu_advanced_incsubtensor1(node):
                 gpu_op = GpuAdvancedIncSubtensor1(
                     set_instead_of_inc=set_instead_of_inc)
             else:
-                gpu_op = GpuAdvancedIncSubtensor1_dev20(
+                # Non deterministic
+                gpu_op = GpuAdvancedIncSubtensor1(
                     set_instead_of_inc=set_instead_of_inc)
             return [gpu_op(as_cuda_ndarray_variable(x),
                            as_cuda_ndarray_variable(y), *coords)]
@@ -1174,7 +1175,7 @@ def local_gpu_advanced_incsubtensor1(node):
                 gpu_op = GpuAdvancedIncSubtensor1(
                     set_instead_of_inc=set_instead_of_inc)
             else:
-                gpu_op = GpuAdvancedIncSubtensor1_dev20(
+                gpu_op = GpuAdvancedIncSubtensor1(
                     set_instead_of_inc=set_instead_of_inc)
             return [host_from_gpu(gpu_op(gpu_x, gpu_y, *coords))]
     return False
