--- train.de
+++ train.de
@@ -16507,7 +16507,6 @@ Ein Mann und eine Frau sitzen auf einer Bank und lesen Bücher.
 Läufer laufen an einer Zuschauermenge vorbei.
 Viele Teilnehmer laufen einen Marathon und die Zuschauer sehen ihnen zu.
 Eine Gruppe Schwimmer springt in ein Schwimmbecken.
-@@
 Ein Sportler schwimmt bei einem Schwimmwettkampf.
 Ein Mann in einem schwarzen Hemd hält ein Papier voller mathematischer Gleichungen hoch.
 Fünf Wanderer gehen einen Hang hinunter.
@@ -16661,7 +16660,6 @@ Ein Mann mit Baseballmütze stellt eine Skulptur eines Mannes her.
 Zwei Männer in weißen Hemden und mit Krawatten nehmen Drinks.
 Ein Mann in einem grün-weiß gestreiften Hemd trägt einen großen Korb auf seiner Schulter.
 Eine Frau in Rot spricht in ein Megaphon, während sie vor einem Spruchband hergeht.
-@@
 Mehrere Personen essen etwas in einem vollen Restaurant.
 Eine Frau überquert die Straße. Die Blickrichtung der Kamera ist zwischen zwei Bussen heraus.
 Ein Publikum beobachtet zwei sitzende Musiker und einen Sprecher, der an einem Podium steht.
--- train.en
+++ train.en
@@ -16507,7 +16507,6 @@ A man and a woman are sitting on a bench reading books.
 Runners race by while a crowd of onlookers watch.
 A large group of people are running a marathon and being watched by spectators.
 A group of swimmers are diving into a swimming pool.
-Front stroke swimming race roped off lap areas.
 An athlete swimming during a swim meet.
 A man in a black shirt, holding up a paper full of math equations.
 Five hikers walk down the hillside.
@@ -16661,7 +16660,6 @@ A man in a baseball cap is creating a sculpture of a man.
 Two men wearing white shirts and ties are having drinks.
 A man in a green and white striped shirt carries a large basket on his shoulder.
 A woman wearing red speaks through a megaphone while walking in front of a banner.
-Young girls weave corn stalks into elaborate designs.
 People are eating food in a busy restaurant.
 A woman is crossing the street, the camera's view is from between two buses.
 An audience are watching two seated musicians and a speaker standing at a podium.
